File: 2020-12-25/sol.go
package main
 
import (
    "fmt"
)

func dis(p float32, d float32) float32 {
    return p-(p*(d/100.0));
}

func dp(a int) int {
    s := 0;
    for(a>0){s += (a%10)*(a%10);a /= 10;}
    return s;
}

func isHappyNumber(n int) string {
    if n==1{
        return "true";
    }
    if n==4 {
        return "false";
    }
    return isHappyNumber(dp(n));
}

func main() {
	fmt.Println(dis(1500, 50));
	fmt.Println(dis(89, 20));
	fmt.Println(dis(100, 75));
    fmt.Println(isHappyNumber(19));
    fmt.Println(isHappyNumber(7));
    fmt.Println(isHappyNumber(32));
    fmt.Println(isHappyNumber(145));
    fmt.Println(isHappyNumber(4));
    fmt.Println(isHappyNumber(89));
    fmt.Println(isHappyNumber(20));
}

File: 2020-12-25/sol.py
from datetime import datetime
from dateutil.relativedelta import relativedelta
 
dis = lambda p,d:p-(p*(d/100))

print(dis(1500,50))
print(dis(89,20))
print(dis(100,75))

def numberOfDays(s : str) -> str:
    diff=relativedelta(datetime.now().date(),datetime.strptime(s, "%m/%d/%Y"))
    return f"{diff.years} year(s), {diff.months} month(s), {diff.days} day(s)"

print(numberOfDays("12/1/2020"))
print(numberOfDays("7/7/2020"))
print(numberOfDays("6/13/2019"))

isHappyNumber = lambda n:eval({1:"True",4:"False"}.get(n,'isHappyNumber(sum(int(x)**2for x in str(n)))'))

print(isHappyNumber(19))
print(isHappyNumber(7))
print(isHappyNumber(32))
print(isHappyNumber(145))
print(isHappyNumber(4))
print(isHappyNumber(89))
print(isHappyNumber(20))
